File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcessResult.java
package org.jetbrains.bsp.bazel.bazelrunner;

import ch.epfl.scala.bsp4j.StatusCode;
import io.vavr.collection.List;
import org.jetbrains.bsp.bazel.bazelrunner.outputs.OutputCollector;
import org.jetbrains.bsp.bazel.commons.ExitCodeMapper;

public class BazelProcessResult {

  private final OutputCollector stdout;
  private final OutputCollector stderr;
  private final int exitCode;

  public BazelProcessResult(OutputCollector stdout, OutputCollector stderr, int exitCode) {
    this.stdout = stdout;
    this.stderr = stderr;
    this.exitCode = exitCode;
  }

  public boolean isNotSuccess() {
    return getStatusCode() != StatusCode.OK;
  }

  public StatusCode getStatusCode() {
    return ExitCodeMapper.mapExitCode(exitCode);
  }

  public List<String> stdoutLines() {
    return stdout.lines();
  }

  public String stdout() {
    return stdout.output();
  }

  public List<String> stderrLines() {
    return stderr.lines();
  }

  public String stderr() {
    return stderr.output();
  }
}

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcessResult.kt
package org.jetbrains.bsp.bazel.bazelrunner

import com.jetbrains.bsp.bsp4kt.StatusCode
import org.jetbrains.bsp.bazel.bazelrunner.outputs.OutputCollector
import org.jetbrains.bsp.bazel.commons.ExitCodeMapper

class BazelProcessResult(
    private val stdoutCollector: OutputCollector,
    private val stderrCollector: OutputCollector,
    private val exitCode: Int
) {
  val isNotSuccess: Boolean get() = statusCode != StatusCode.Ok
  val statusCode: StatusCode get() = ExitCodeMapper.mapExitCode(exitCode)
  val stdoutLines: List<String> get() = stdoutCollector.lines()
  val stdout: String get() = stdoutCollector.output()
  val stderrLines: List<String> get() = stderrCollector.lines()
  val stderr: String get() = stderrCollector.output()
}
